lib/classes/batch/class-batch-task.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Batch Task 
+ *
+ * @since 4.0.0
+ */
+
+namespace wpCloud\StatelessMedia\Batch;
+
+abstract class BatchTask implements IBatchTask {
+  /**
+   * Current state of the task
+   * 
+   * @var array
+   */
+  protected $state = [];
+
+  /**
+   * ID of the task
+   * 
+   * @var string
+   */
+  protected $id = 'batch_task';
+
+  /**
+   * Description of the task (for displaying status)
+   * 
+   * @var string
+   */
+  protected $description = '';
+
+  /**
+   * Total number of items to process. Default -1 - infinite/unknown
+   * 
+   * @var int
+   */
+  protected $total = -1;
+
+  /**
+   * Number of items processed
+   * 
+   * @var int
+   */
+  protected $completed = 0;
+
+  /**
+   * Number of items to process in a single batch
+   * 
+   * @var int
+   */
+  protected $limit = 20;
```

Comment:
@balexey88 new CONSTANT?

Reply:
This is basically just sample value. In real life the `limit` will depend on the task itself and should be re-declared in each descendant.